### DISCUSSION TITLE
Change charger power to float from integer

### DISCUSSION THREAD

**Discussion by carryonrewardless** [alandtse/tesla#859] [Ideas]
I have no idea how to create a pull request so anyone so inclined can do this fairly basic request change.

Change this
```
     @property
    def native_value(self) ->int:
        """Return the charger power."""
        return self._car.charger_power
```
to this
```
     @property
    def native_value(self) ->float:
        """Return the charger power."""
        return self._car.charger_power
``` 
at line 188 in https://github.com/alandtse/tesla/blob/dev/custom_components/tesla_custom/sensor.py

to output charger_power as a float instead of an integer.

**Comment by purcell-lab** [comment]
What are the advantages of providing this as a float?

It won't improve the accuracy, Tesla only provide this as an int.

**Reply by carryonrewardless** [reply]
The advantage comes if Tesla provide it as a float.  Is that likely to happen?

**Reply by purcell-lab** [reply]
You are correct, with the FleetAPI and telemetry streaming Tesla are providing charger power as a float.
![image](https://github.com/alandtse/tesla/assets/79175134/e6d565e6-b1bf-4a3c-b50b-8018c37f6619)
